### PR TITLE
Use interpolated tags syntax to add extra tags to cluster nodes

### PR DESCRIPTION
**Pull request by sclausson**
https://github.com/hashicorp/terraform-aws-vault/issues/7

### CODE REVIEW

File: modules/vault-cluster/variables.tf [hunk 1]
```diff
@@ -76,8 +76,13 @@ variable "allowed_ssh_security_group_ids" {
...
 variable "cluster_tag_key" {
-  description = "Add a tag with this key and the value var.cluster_name to each Instance in the ASG."
-  default     = "Name"
+  description = "Add a tag with this key and the value var.cluster_tag_value to each Instance in the ASG."
+  default     = "vault-servers"
+}
+
+variable "cluster_tag_value" {
+  description = "Add a tag with this value and the key var.cluster_tag_key to each Instance in the ASG."
+  default     = "auto-join"
```
Comment (brikis98) [inline modules/vault-cluster/variables.tf]: Looking at this with fresh eyes, it seems like the approach we took with tags is a little too hard-coded. A better approach may be to take advantage of the [tags](https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#tags) parameter on ASGs. Something along the lines of:

```hcl
variable "tags" {
  type = "list"
  # Example:
  # default = {
  #   {
  #     key = "foo"
  #     value = "bar"
  #     propagate_at_launch = true  
  #   }
  # }
}

resource "aws_autoscaling_group" "vault" {
  # ...
  tags = "${merge(map("Name", var.cluster_name), var.tags)}"
}
```

This way, users can set any arbitrary tags they want, including overriding the `Name` if they so choose.

Comment (sclausson) [reply modules/vault-cluster/variables.tf]: @brikis98 I agree this is a much better solution. I'll made the changes in my PR.

**Comment by sclausson** [comment]
@brikis98 I've amended this commit to support the interpolated tags syntax.

File: modules/vault-cluster/variables.tf [hunk 2]
```diff
@@ -80,6 +80,12 @@ variable "cluster_tag_key" {
   default     = "Name"
 }
 
+variable "cluster_extra_tags" {
+  description = "A list of additional tags to add to each Instance in the ASG."
+  type = "list"
+  default = []
```
Comment (brikis98) [inline modules/vault-cluster/variables.tf]: Could you update the `description` to mention that each element in the list should be a map with the keys `key`, `value`, and `propagate_at_launch`? Also, an example in comments above `default` would be helpful.

**Comment by sclausson** [comment]
Ok should be all set now with requested changes.

**Comment by brikis98** [comment]
https://github.com/hashicorp/terraform-aws-vault/releases/tag/v0.0.6

**Pull request merged**